# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/nsBaseWidget.cpp
# Commit: 3dd35da1c8be
# Full Hash: 3dd35da1c8bee1353c935e045e7518bc01ea3e77
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-05-09 21:30:55
# Description:
#   Bug 1635769 - Call WebRenderLayerManager::ClearCachedResources() during re-creating widget r=tnikkel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D74392
# ==============================================================================

diff -r efb86c809f5c -r 3dd35da1c8be widget/nsBaseWidget.cpp
--- a/widget/nsBaseWidget.cpp	Sat May 09 09:38:52 2020 +0300
+++ b/widget/nsBaseWidget.cpp	Sat May 09 05:44:37 2020 +0000
@@ -1382,6 +1382,13 @@
   return mCompositorBridgeChild;
 }
 
+void nsBaseWidget::ClearCachedWebrenderResources() {
+  if (!mLayerManager || !mLayerManager->AsWebRenderLayerManager()) {
+    return;
+  }
+  mLayerManager->ClearCachedResources();
+}
+
 already_AddRefed<gfx::DrawTarget> nsBaseWidget::StartRemoteDrawing() {
   return nullptr;
 }